pennylane_lightning/core/bindings/Bindings_nb.hpp

```diff
@@ -189,12 +188,10 @@ auto alignedArray(Util::CPUMemoryModel memory_model, std::size_t size,
     auto capsule =
         nb::capsule(ptr, [](void *p) noexcept { Util::alignedFree(p); });
 
-    // Create shape array
-    size_t ndim = 1;
-    size_t shape[ndim] = {size};
+    std::vector<size_t> shape{size};
 
     // Return ndarray with custom allocated memory
-    return nb::ndarray<T>(ptr, ndim, shape, capsule);
+    return nb::ndarray<T, nb::c_contig>(ptr, 1, shape.data(), capsule);
```

Comment:
```suggestion
    return nb::ndarray<T, nb::c_contig>(ptr, shape.size(), shape.data(), capsule);
```

To help avoid magic numbers (this is minor, feel free to overrule me on this)

Reply:
(Same comment applies to a few places)